nexus/src/app/metrics.rs

```diff
@@ -39,4 +55,45 @@ impl super::Nexus {
         )
         .await
     }
+
+    pub async fn silo_metric_list(
+        &self,
+        opctx: &OpContext,
+        metric_name: SystemMetricName,
+        project_lookup: Option<lookup::Project<'_>>,
+        pagination: PaginationParams<ResourceMetrics, ResourceMetrics>,
+        limit: NonZeroU32,
+    ) -> Result<dropshot::ResultsPage<Measurement>, Error> {
+        let authz_silo =
+            opctx.authn.silo_required().internal_context("listing metrics")?;
+        // must be a silo reader to use this path at all
+        opctx.authorize(authz::Action::Read, &authz_silo).await?;
+
+        let resource_id = match project_lookup {
+            Some(project_lookup) => {
+                // make sure project exists and current user can read it
+                let (.., authz_project) =
+                    project_lookup.lookup_for(authz::Action::Read).await?;
+                authz_project.id()
+            }
+            // if no project specified, scope to current silo
+            _ => authz_silo.id(),
+        };
+
+        let timeseries = match metric_name {
+            SystemMetricName::VirtualDiskSpaceProvisioned
+            | SystemMetricName::CpusProvisioned
+            | SystemMetricName::RamProvisioned => {
+                opctx.authorize(authz::Action::Read, &authz_silo).await?;
+                format!("collection_target:{metric_name}")
+            }
+        };
+        self.select_timeseries(
+            &timeseries,
+            &[&format!("id=={}", resource_id)],
+            pagination,
+            limit,
+        )
+        .await
+    }
```

Comment:
@davepacheco any thoughts on the authz checks here, especially how to test them? I don't think the verify endpoints test is sufficient.